### ISSUE TITLE
[BUG] XML attribute incompatible with JSON $ref

### ISSUE THREAD

**Issue by ghilainm** [OpenAPITools/openapi-generator#3178]
##### Description

When using reference to describe property, the xml meta information is lost in the parsing.
It is therefore impossible when using reference to define xml related information. This seems like a big limitation.

##### openapi-generator version

3.0.2

##### OpenAPI declaration file content or url

<!-- if it is a bug, a json or yaml that produces it.
If you post the code inline, please wrap it with
```yaml
openapi: 3.0.2
paths: {}
components:
  schemas:
    XXX:
      type: object
      properties:
        MyAttribute:
          $ref: "#/components/schemas/MyAttributeType"
          xml:
            attribute: true
            name: MyAttributeSpecialName
```

##### Suggest a fix

No idea as when using $ref it is supposed to replace the block content, therefore I don't know how this should be handled.

**Comment by wing328** [comment]
I don't think this is supported. As a workaround, please keep it simple by defining the type inline.

**Comment by ghilainm** [comment]
@wing328 I can't, OpenAPI specifications are generated from XSD on my side :). And latter on I don't want to have two java classes generated for the same type. So this is not really a s solution.

What is imposing this restriction ? JSON schema or OpenAPI?

**Comment by slogic** [comment]
@ghilainm Did you generate opepapi schema by hand or by some tools? It latter please provide some info. Thanx. Concerning topicstarter issue, meta info is ignored. You need to wrap it into allOf if you can't bundle into schema.
